platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -175,7 +175,21 @@ class HangingProtocolService {
   public getProtocolById(id: string): HangingProtocol.Protocol {
     const protocol = this.protocols.get(id);
 
-    return protocol;
+    if (protocol instanceof Function) {
```

Comment:
Does it make sense to instead check for ```protocol instanceof HangingProtocol.ProtocolGenerator```?

Reply:
I think the "instanceof" will fail. ProtocolGenerator is a type not a class where we create an instance from. In the "getHangingProtocolModule", the protocol generator is defined as a standard function but conforms to this type. 
Not sure if there is a better way to check the method signature is valid before execution

Reply:
I think you are correct. There is not a better way.